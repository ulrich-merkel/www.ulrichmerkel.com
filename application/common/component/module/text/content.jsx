/* eslint-disable immutable/no-mutation */
/**
 * Es6 module for React Component.
 * Component module React classes combine elements to
 * bigger parts of the page.
 *
 * @file
 * @module
 * @flow weak
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2016
 * @version 0.0.1
 *
 * @requires react
 * @requires prop-types
 * @requires classnames
 * @requires shortid
 * @requires common/component/element/paragraph
 *
 * @changelog
 * - 0.0.1 Basic functions and structure
 */
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import shortid from 'shortid';

import P from '../../element/paragraph';

/**
 * Function representing a component to return a single react child element.
 *
 * @function
 * @param {Object} [props] - The current component props
 * @param {Array<string>} [props.content] - The content's text
 * @param {boolean} [props.hasColumns2=false] - Whether the component text should be clusted in columns via css or not
 * @param {boolean} [props.isCentered=false] - Whether the component text should be centered via css or not
 * @returns {ReactElement|null} React component markup
 */
function ModuleTextContent(props) {
    const {
        content,
        hasColumns2,
        isCentered
    } = props;

    if (!Array.isArray(content) || !content.length) {
        return null;
    }

    const componentTextContentClassName = classnames(
        {
            'has-columns--2': hasColumns2,
            'is-centered': isCentered
        },
        'm-text__content'
    );

    return (
        <div className={componentTextContentClassName}>
            {content.map(function (html) {
                return (
                    <P key={shortid.generate()} dangerouslySetInnerHTML={{ __html: html }} />
                );
            })}
        </div>
    );
}

/**
 * Validate props via React.PropTypes helpers.
 *
 * @static
 * @type {Object}
 */
ModuleTextContent.propTypes = {
    content: PropTypes.arrayOf(PropTypes.string), // eslint-disable-line react/require-default-props
    hasColumns2: PropTypes.bool,
    isCentered: PropTypes.bool
};

/**
 * Set defaults if props aren't available.
 *
 * @static
 * @type {Object}
 */
ModuleTextContent.defaultProps = {
    hasColumns2: false,
    isCentered: false
};

export default ModuleTextContent;
