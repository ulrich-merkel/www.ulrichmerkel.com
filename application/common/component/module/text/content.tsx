/**
 * Es6 module for React Component.
 * Component module React classes combine elements to
 * bigger parts of the page.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { default as React, FunctionComponent } from 'react';
import classnames from 'classnames';
import shortid from 'shortid';

import { P } from '../../element/paragraph';

type Props = {
    content?: string[];
    hasColumns2?: boolean;
    isCentered?: boolean;
}

/**
 * Function representing a component to return a single react child element.
 *
 * @function
 * @param {object} [props] - The current component props
 * @param {Array<string>} [props.content] - The content's text
 * @param {boolean} [props.hasColumns2=false] - Whether the component text should be clusted in columns via css or not
 * @param {boolean} [props.isCentered=false] - Whether the component text should be centered via css or not
 * @returns {ReactElement|null} React component markup
 */
export const ModuleTextContent: FunctionComponent<Props> = (props) => {
    const { content, hasColumns2 = false, isCentered = false } = props;

    if (!Array.isArray(content) || !content.length) {
        return null;
    }

    const componentTextContentClassName = classnames(
        {
            'has-columns--2': hasColumns2,
            'is-centered': isCentered
        },
        'm-text__content'
    );

    return (
        <div className={componentTextContentClassName}>
            {content.map(function mapContent(html) {
                return (
                    <P
                        key={shortid.generate()}
                        dangerouslySetInnerHTML={{ __html: html }}
                    />
                );
            })}
        </div>
    );
}
