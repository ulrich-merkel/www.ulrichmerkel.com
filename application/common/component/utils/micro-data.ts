/**
 * Es6 module for creating mirco data attributes.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isValidString } from '../../utils/string';

type ItemTypeAttributes = {
    itemScope: boolean;
    itemType: string;
};

/**
 * Get component properties for itemProp and itemType.
 *
 * @param {boolean} itemType - The itemType property if available
 * @returns {object|null} The component props for spreading
 */
export function getItemTypeAttributes(
    itemType?: string
): ItemTypeAttributes | null {
    return isValidString(itemType)
        ? {
              itemScope: true,
              itemType
          }
        : null;
}
