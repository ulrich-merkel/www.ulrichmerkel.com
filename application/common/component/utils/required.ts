/**
 * Es6 module for creating form input required attributes.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isBoolean } from 'lodash';

type RequiredAttributes = {
    'aria-required': boolean;
    required: boolean;
};

/**
 * Get component properties for required form input elements.
 *
 * @param {boolean} required - Whether the input element is required or not
 * @returns {object|null} The component props for spreading
 */
export function getRequiredAttributes(
    required?: boolean
): RequiredAttributes | null {
    return isBoolean(required) && required
        ? {
              'aria-required': true,
              required: true
          }
        : null;
}
