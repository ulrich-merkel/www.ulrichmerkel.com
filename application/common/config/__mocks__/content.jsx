const mockedConfigContent = {
    PageIndex: {
        head: {
            title: 'intl-title-1',
            meta: [
                {
                    name: 'description',
                    content: 'intl-content-1'
                }
            ]
        },
        section1: {
            headline: 'intl-headline-1',
            lead: 'intl-lead-1',
            btnTitle: 'intl-title-1',
            btnLabel: 'intl-label-1'
        }
    },
    PagePersona: {
        head: {
            title: 'intl-title-2',
            meta: [
                {
                    name: 'description',
                    content: 'intl-content-2'
                }
            ]
        },
        section1: {
            headline: 'intl-headline-2',
            lead: 'intl-lead-2',
            btnTitle: 'intl-title-2',
            btnLabel: 'intl-label-2',
            text: [
                {
                    content: [
                        'intl-text-2'
                    ]
                }
            ]
        }
    }
};

export default mockedConfigContent;
