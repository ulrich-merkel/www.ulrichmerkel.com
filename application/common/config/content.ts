/**
 * Handle config.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { configHead } from './content/head';
import { configLayoutDialog } from './content/layout-dialog';
import { configLayoutFooter } from './content/layout-footer';
import { configLayoutHeader } from './content/layout-header';
import { configLayoutLoader } from './content/layout-loader';
import { configPageBroadcast } from './content/page-broadcast';
import { configPageContact } from './content/page-contact';
import { configPageDisclaimer } from './content/page-disclaimer';
import { configPageHome } from './content/page-home';
import { configPageImprint } from './content/page-imprint';
import { configPageNotFound } from './content/page-not-found';
import { configPagePersona } from './content/page-persona';
import { configPagePrivacy } from './content/page-privacy';
import { configPageSearch } from './content/page-search';
import { configPageSettings } from './content/page-settings';
import { configPageWork } from './content/page-work';
import { configPageWorkGedankenKollektiv } from './content/page-work-gedanken-kollektiv';
import { configPageWorkLebensweltSchule } from './content/page-work-lebenswelt-schule';
import { configPageWorkLikeArchitecture } from './content/page-work-like-architecture';
import { configPageWorkMomentariness } from './content/page-work-momentariness';
import { configPageWorkOptikLudewig } from './content/page-work-optik-ludewig';
import { configPageWorkRevolution } from './content/page-work-revolution';
import { configPageWorkSummerInspiration } from './content/page-work-summer-inspiration';
import { configPageWorkVerlegeserviceBunge } from './content/page-work-verlegeservice-bunge';

export const configContent = {
    Head: configHead,
    LayoutDialog: configLayoutDialog,
    LayoutFooter: configLayoutFooter,
    LayoutHeader: configLayoutHeader,
    LayoutLoader: configLayoutLoader,
    PageBroadcast: configPageBroadcast,
    PageContact: configPageContact,
    PageDisclaimer: configPageDisclaimer,
    PageHome: configPageHome,
    PageImprint: configPageImprint,
    PageNotFound: configPageNotFound,
    PagePersona: configPagePersona,
    PagePrivacy: configPagePrivacy,
    PageSearch: configPageSearch,
    PageSettings: configPageSettings,
    PageWork: configPageWork,
    PageWorkGedankenKollektiv: configPageWorkGedankenKollektiv,
    PageWorkLebensweltSchule: configPageWorkLebensweltSchule,
    PageWorkLikeArchitecture: configPageWorkLikeArchitecture,
    PageWorkMomentariness: configPageWorkMomentariness,
    PageWorkOptikLudewig: configPageWorkOptikLudewig,
    PageWorkRevolution: configPageWorkRevolution,
    PageWorkSummerInspiration: configPageWorkSummerInspiration,
    PageWorkVerlegeserviceBunge: configPageWorkVerlegeserviceBunge
};
