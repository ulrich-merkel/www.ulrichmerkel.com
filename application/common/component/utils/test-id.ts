/**
 * Es6 module for creating test id related attributes
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isValidString } from '../../utils/string';

type TestIdAttributes = {
    'data-testid': string;
};

/**
 * Get component properties for testing.
 *
 * @param {string} testId - The test id to be added
 * @returns {object|null} The component props for spreading
 */
export function getTestIdAttributes(testId?: string): TestIdAttributes | null {
    return isValidString(testId)
        ? {
              'data-testid': testId
          }
        : null;
}
