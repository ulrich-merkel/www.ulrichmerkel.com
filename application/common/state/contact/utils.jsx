/**
 * Es6 module for Redux Architecture.
 * Simple helper component to add utility functions.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2016
 * @version 0.0.1
 *
 * @see {@link http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html}
 *
 * @TODO Should be refactored and moved to actions
 *
 * @requires common/utils/validation
 *
 * @changelog
 * - 0.0.1 Basic functions and structure
 */
import { required, email } from '../../utils/validation';
import { isBrowser } from '../../utils/environment';

/**
 * Validation helper to check all inputs.
 *
 * @function
 * @param {object} [state={}] - The current redux contact state if given
 * @returns {object} The validation status for all inputs as object
 */
function validate(state = {}) {
    return {
        name: required(state.name),
        email: required(state.email) && email(state.email),
        subject: required(state.subject),
        message: required(state.message)
    };
}

/**
 * Validation single or all input element(s), depending on parameter input.
 *
 * @function
 * @param {object} state - The current redux contact state
 * @returns {boolean} The validation status
 */
function isValid(state) {
    const validated = validate(state);

    return !Object.keys(validated).some(function someKey(key) {
        return validated[key] !== undefined && !validated[key];
    });
}

function canSendForm(state) {
    const isStateValid = isValid(state);

    return isStateValid
        && state.pristine
        && !state.pending
        && isBrowser(state);
}

export {
    canSendForm,
    isValid,
    validate
};
