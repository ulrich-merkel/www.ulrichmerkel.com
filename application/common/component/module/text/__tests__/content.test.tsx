import * as React from 'react';
import { render } from '../../../../../__tests__/utils/test-utils';
import { ModuleTextContent } from '../content';

describe('ModuleTextContent', function fnDescribe() {
    const defaultProps = {
        content: ['Module text headline', 'Module text subline'],
        hasColumns2: true,
        isCentered: true
    };

    it('should render correctly', function fnIt() {
        const { asFragment } = render(<ModuleTextContent {...defaultProps} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should return null if content is empty', function fnIt() {
        const { asFragment } = render(
            <ModuleTextContent {...defaultProps} content={null} />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
