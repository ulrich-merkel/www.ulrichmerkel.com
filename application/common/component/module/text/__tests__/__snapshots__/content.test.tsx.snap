// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleTextContent should render correctly 1`] = `
<DocumentFragment>
  <div
    class="has-columns--2 is-centered m-text__content"
  >
    <p
      class="c-type--p"
    >
      Module text headline
    </p>
    <p
      class="c-type--p"
    >
      Module text subline
    </p>
  </div>
</DocumentFragment>
`;

exports[`ModuleTextContent should return null if content is empty 1`] = `<DocumentFragment />`;
